### PR TITLE
Periodic validations

### PR DESCRIPTION
**Pull request by micapam**
Added a `period` attribute on Dataset. Valid values so far: `free`, `month` (defaults to `month`).

If `free`, datapoints for that dataset can be added with any `ts` value (as now).

If `month`, validation on the datapoints will ensure that any datapoint added is not in the same month as any existing datapoint. 

We could extend this in obvious ways for week, day, hour etc.

### CODE REVIEW

File: app/models/datapoint.rb [hunk 1]
```diff
@@ -5,11 +5,27 @@ class Datapoint < ApplicationRecord
...
+  scope :of_year, -> (year) { where('EXTRACT(YEAR FROM ts) = ?', year) }
 
   def label
     ts.strftime("%Y-%m")
   end
 
+  private
+
+  def check_frequency
+    if dataset&.period.present?
```
Comment (tobyhede) [inline app/models/datapoint.rb]: That is nifty

File: app/models/dataset.rb [hunk 2]
```diff
@@ -2,6 +2,8 @@ class Dataset < ApplicationRecord
   include Measurable
   include Nameable
 
+  PERIODS = %w(month week day)
```
Comment (tobyhede) [inline app/models/dataset.rb]: Is it worth setting a default?

Comment (micapam) [inline app/models/dataset.rb]: Currently default is nil (which is a valid value) - do you think it should be `month`? For instance - would all our current datasets be monthly?

**Comment by micapam** [comment]
Good point re. the import data - I will do that - or set a default as you mentioned (same result).

File: app/models/dataset.rb (hunk 2 above)
Comment (tobyhede) [inline app/models/dataset.rb]: Everything is monthly at the moment. 

I think. 

Pretty sure. 

cc @elisechant ?

Comment (elisechant) [inline app/models/dataset.rb]: yes monthly or no period

**Comment by micapam** [comment]
can haz merge ?

**Comment by tobyhede** [comment]
:shipit:

**Pull request merged**